### PR TITLE
Cascade delete for applicant and position to delete related interviews 

### PR DESCRIPTION
**Pull request by likeabowx**
I have added in a method to delete the related interviews, under `DeleteApplicantCommand.java` and `DeletePositionCommand.java` when deleting applicant and position respectively.

It works well on my side and I have included in the message to show how many interviews are deleted as well.

image

### CODE REVIEW

**Comment by likeabowx** [comment]
The test cases are failing because I changed the expected message in `CommandResult`, but I am not very sure how to update the test case as the deleted interviews count can only be obtained after executing the command.

**Comment by likeabowx** [comment]
> The test cases are failing because I changed the expected message in `CommandResult`, but I am not very sure how to update the test case as the deleted interviews count can only be obtained after executing the command.

I managed to fix the test cases by editing the expected message.

File: src/main/java/seedu/address/logic/commands/applicant/DeleteApplicantCommand.java [hunk 1]
```diff
@@ -12,6 +14,7 @@
 import seedu.address.logic.commands.exceptions.CommandException;
 import seedu.address.model.Model;
 import seedu.address.model.applicant.Applicant;
+import seedu.address.model.interview.Interview;
 
 /**
  * Deletes a applicant identified using it's displayed index from the address book.
```
Comment (goalfix) [inline src/main/java/seedu/address/logic/commands/applicant/DeleteApplicantCommand.java:20]: Would it be clearer to add `Will delete any interviews that the applicant has as well` for this javadoc comment?

File: src/main/java/seedu/address/logic/commands/applicant/DeleteApplicantCommand.java [hunk 2]
```diff
@@ -41,11 +46,37 @@ public CommandResult execute(Model model) throws CommandException {
...
         model.deletePerson(applicantToDelete);
-        return new CommandResult(String.format(MESSAGE_DELETE_PERSON_SUCCESS, applicantToDelete),
+        return new CommandResult(
+                String.format(MESSAGE_DELETE_PERSON_SUCCESS, applicantToDelete) + "\n"
+                        + String.format(MESSAGE_DELETE_INTERVIEWS, deleteInterviewCount),
                 getCommandDataType());
     }
 
+    /**
+     * Deletes interviews which are for the applicant to delete.
```
Comment (goalfix) [inline src/main/java/seedu/address/logic/commands/applicant/DeleteApplicantCommand.java:59]: Would it be clearer to add in the return value as well? Maybe something like `@return number of interviews deleted`

File: src/main/java/seedu/address/logic/commands/position/DeletePositionCommand.java [hunk 3]
```diff
@@ -38,11 +43,36 @@ public CommandResult execute(Model model) throws CommandException {
...
+
         model.deletePosition(positionToDelete);
-        return new CommandResult(String.format(MESSAGE_DELETE_POSITION_SUCCESS, positionToDelete),
+        return new CommandResult(String.format(MESSAGE_DELETE_POSITION_SUCCESS, positionToDelete) + "\n"
+                    + String.format(MESSAGE_DELETE_INTERVIEWS, deleteInterviewCount),
                 getCommandDataType());
     }
 
+    /**
+     * Deletes interviews which are for the position to delete.
```
Comment (goalfix) [inline src/main/java/seedu/address/logic/commands/position/DeletePositionCommand.java:55]: Similar to above comment

**Review by goalfix — APPROVED** [review]
Minor changes requested on javadocs to improve clarity. Otherwise, this feature is critical and will help prevent bugs. LGTM!

**Pull request merged**